src/main/resources/config/liquibase/changelog/20220824120842_changelog.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">
+    <changeSet author="ge65cer" id="20220824120842">
+        <comment>Add a table to persist the user progress in learning goals.</comment>
+        <createTable tableName="learning_goal_user">
+            <column name="learning_goal_id" type="BIGINT">
+                <constraints nullable="false" primaryKey="true"/>
+            </column>
+            <column name="user_id" type="BIGINT">
+                <constraints nullable="false" primaryKey="true"/>
+            </column>
+            <column name="progress" type="DOUBLE"/>
+            <column name="confidence" type="DOUBLE"/>
+            <column name="last_modified_date" type="TIMESTAMP"/>
+        </createTable>
+
+        <createIndex tableName="learning_goal_user" indexName="learning_goal_progress_modified_date">
+            <column name="last_modified_date" />
+        </createIndex>
+
+        <addUniqueConstraint columnNames="learning_goal_id, user_id" constraintName="uc_learning_goal_user" tableName="learning_goal_user" />
+
+        <addForeignKeyConstraint baseColumnNames="learning_goal_id" baseTableName="learning_goal_user"
+                                 constraintName="FKba695bf5823818d0997937bc5" deferrable="false"
```

Comment:
❓Maybe use a human readable FK name here?

Reply:
We don't seem to have a consistent convention here. I'll keep it for now.